samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -323,7 +305,93 @@ private static void validateIntermediateStreamPartitions(JobGraph jobGraph) {
     }
   }
 
+  /**
+   * Ensures that all streams in supplied {@link JoinedStreamsGroup} agree in partition count.
+   * This includes setting partition counts of any intermediate streams that do not have their
+   * partition counts set.
+   */
+  private static void validateJoinedStreamsGroupPartitions(JoinedStreamsGroup joinedStreamsGroup) {
```

Comment:
Is it the case where inference and validation need to coexist? 
If yes, can we rename this method to something reflective instead of `validateJoinedStreamsGroupPartitioms` which seems to suggest this is a non-mutative method?

Reply:
I agree. I'll try to think of something and I'm also open to suggestions.